pkg/web/vuex/vuex.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Cohorse Author
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vuex
+
+import (
+	"fmt"
+	"syscall/js"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// Store is a Vuex store which can be used to dispatch actions or commit
+// mutations. It stores the underlying JS object and all actions which are
+// implemented in Go and can be dispatched using the `CohorseVuexDispatch` from
+// JS.
+type Store struct {
+	o       js.Value
+	actions map[string]Action
+}
+
+// StoreIndex is the index of all known stores.
+type StoreIndex map[string]Store
+
+// ActionIndex is the index of all known stores. Actions are scoped per store.
```

Comment:
Is this comment correct? Or is it the index of actions?

Reply:
This should be fine. We have two index types in Go.
- StoreIndex: The index that holds all available stores (global + individual store modules in the future)
- ActionIndex: Simply a list of actions that a store is known to support. Two stores could in theory also support the same set of actions from the same action index.